test/DSRMainnetOracle.t.sol

```diff
@@ -0,0 +1,238 @@
+
+        oracle.getConversionRateBinomialApprox(block.timestamp + 1 hours);
+    }
+
+    function test_gas_getConversionRateBinomialApprox_1year() public {
+        vm.pauseGasMetering();
+        pot.setDSR(FIVE_PCT_APY_DSR);
+        pot.setChi(1.03e27);
+        pot.setRho(block.timestamp);
+        oracle.refresh();
+        vm.resumeGasMetering();
+
+        oracle.getConversionRateBinomialApprox(ONE_YEAR);
+    }
+
+    function test_getConversionRateBinomialApprox_pastRevert() public {
+        uint256 rho = oracle.getRho();
+        vm.expectRevert("DSROracleBase/invalid-timestamp");
+        oracle.getConversionRateBinomialApprox(rho - 1);
+
+        oracle.getConversionRateBinomialApprox(rho);
+    }
+
+    function test_getConversionRateLinearApprox() public {
+        assertEq(oracle.getConversionRateLinearApprox(),         1e27);
+        assertEq(oracle.getConversionRateLinearApprox(ONE_YEAR), 1e27);
+
+        pot.setDSR(FIVE_PCT_APY_DSR);
+        pot.setChi(1.03e27);
+        pot.setRho(block.timestamp);
+        oracle.refresh();
+
+        assertEq(oracle.getConversionRateLinearApprox(),         1.03e27);
+        assertEq(oracle.getConversionRateLinearApprox(ONE_YEAR), 1.080253869133389495792931840e27);   // 5% interest on 1.03 value = 1.0815, but linear approx is 1.0802
+    }
+
+    function test_gas_getConversionRateLinearApprox_1hour() public {
+        vm.pauseGasMetering();
+        pot.setDSR(FIVE_PCT_APY_DSR);
+        pot.setChi(1.03e27);
+        pot.setRho(block.timestamp);
+        oracle.refresh();
+        vm.resumeGasMetering();
+
+        oracle.getConversionRateLinearApprox(block.timestamp + 1 hours);
+    }
+
+    function test_gas_getConversionRateLinearApprox_1year() public {
+        vm.pauseGasMetering();
+        pot.setDSR(FIVE_PCT_APY_DSR);
+        pot.setChi(1.03e27);
+        pot.setRho(block.timestamp);
+        oracle.refresh();
+        vm.resumeGasMetering();
+
+        oracle.getConversionRateLinearApprox(ONE_YEAR);
+    }
+
+    function test_getConversionRateLinearApprox_pastRevert() public {
+        uint256 rho = oracle.getRho();
+        vm.expectRevert("DSROracleBase/invalid-timestamp");
+        oracle.getConversionRateLinearApprox(rho - 1);
+
+        oracle.getConversionRateLinearApprox(rho);
+    }
+
+    function test_binomialAccuracyLongDuration() public {
+        pot.setDSR(FIVE_PCT_APY_DSR);
+        pot.setChi(1.03e27);
+        pot.setRho(block.timestamp);
+        oracle.refresh();
+
+        // Even after a year the binomial is accurate to within 0.001%
+        assertApproxEqRel(
+            oracle.getConversionRate(ONE_YEAR),
+            oracle.getConversionRateBinomialApprox(ONE_YEAR),
+            0.00001e18
+        );
+    }
+
+    function test_getConversionRateFuzz(uint256 rate, uint256 duration) public {
+        rate     = bound(rate,     0, 0.5e27);  // Bound by 0-50% APR
+        duration = bound(duration, 0, 1 days);  // Bound by 1 day
```

Comment:
Lets make this longer than a day, maybe 30 days

Reply:
ok extended